rust/src/parser.rs

```diff
@@ -16,6 +15,14 @@ use nom::{
 use std::collections::HashMap;
 
 // AST expression types for MistQL
+#[derive(Debug, Clone, PartialEq)]
+pub enum LiteralValue {
```

Comment:
NB: Could call this LiteralPrimitive, since technically `[foo, bar]` is considered an array literal.